src/lib.rs

```diff
@@ -15,28 +16,35 @@ pub use session::{RequestSession, SessionMiddleware};
 
 mod session;
 
-pub struct Middleware {
-    key: Vec<u8>,
-}
+#[derive(Default)]
+pub struct Middleware {}
 
 impl Middleware {
-    pub fn new(key: &[u8]) -> Middleware {
-        Middleware { key: key.to_vec() }
+    pub fn new() -> Self {
+        Default::default()
     }
 }
 
+fn parse_pair(key_value: &str) -> Option<(String, String)> {
```

Comment:
Not sure if it's possible to get rid of allocations (String).
jar.add_original expects Cookie<'static>, which means, as I understand, we can only use either &'static str or String for key and value.
But somehow it worked before.